core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -368,6 +373,40 @@ public AbstractClientOptionsBuilder followRedirects(RedirectConfig redirectConfi
         return option(REDIRECT_CONFIG, requireNonNull(redirectConfig, "redirectConfig"));
     }
 
+    /**
+     * Adds the specified {@link ClientRequestContext} customization function so that
+     * the customized context is used when the client invokes a request from the current thread.
+     * <pre>{@code
+     * Clients.builder(...)
+     *        .contextCustomizer(ctx -> {
+     *            // Get traceId from the current thread local storage.
+     *            String traceId = Trace.current().currentTraceContext()...;
+     *            ctx.setAttr(TRACE_ID, traceId);
+     *        })
+     *        .build();
```

Comment:
Please make this not use Trace but just a generic thread local variable so people who are not familiar with distributed tracing can understand it. Also, please add:

- a line that actually 'initiates' the call to make it clear that the customizer will be invoked from the current thread
- a counter-example that initiates the call from other thread (which won't work.); and
- update the `Clients.withContextCustomizer()` Javadoc as well.